pkg/util/ecs/metadata/clients.go

```diff
@@ -92,28 +92,28 @@ func V2() (*v2.Client, error) {
 	return globalUtil.v2, nil
 }
 
-// V3FromCurrentTask returns a client for the ECS metadata API v3 by detecting
+// V3orV4FromCurrentTask returns a client for the ECS metadata API v3 or v4 by detecting
 // the endpoint address from the task the executable is running in. Returns an
 // error if it was not possible to detect the endpoint address.
-func V3FromCurrentTask() (*v3.Client, error) {
+func V3orV4FromCurrentTask() (*v3or4.Client, error) {
```

Comment:
Could we document that v4 is prioritized over v3? and explain why?

Reply:
Done in https://github.com/DataDog/datadog-agent/pull/10727/commits/06fc659a00eeff102e617637170ec31d7483fc30#diff-c24024a2d89ee0b600ac5251b8fd26fb42b17c290897bc408f9292e3e9f8a75aR98